### PR TITLE
rename return types of some Get operations

### PR DESCRIPTION
**Pull request by Jinming-Hu**
- Changed return type of `BlobServiceClient::GetUserDelegationKey` to `UserDelegationKey`.
- Changed return type of `BlobServiceClient::GetProperties` to `BlobServiceProperties`.
- Changed return type of `BlobServiceClient::GetAccountInfo` to `AccountInfo`.
- Changed return type of `BlobServiceClient::GetStatistics` to `ServiceStatistics`.
- Changed return type of `BlobContainerClient::GetProperties` to `BlobContainerProperties`.
- Changed return type of `BlobContainerClient::GetAccessPolicy` to `BlobContainerAccessPolicy`.
- Changed return type of `BlobClient::GetProperties` to `BlobProperties`.

I didn't change the return type of `BlockBlobClient::GetBlockList` and `PageBlobClient::GetPageRanges`

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [ ] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Comment by Jinming-Hu** [comment]
blocked by https://github.com/Azure/azure-sdk-for-cpp/issues/1828

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp [hunk 1]
```diff
@@ -239,9 +239,9 @@ namespace Azure { namespace Storage { namespace Blobs {
      *
      * @param options Optional parameters to execute this function.
      * @param context Context for cancelling long running operations.
-     * @return A GetBlobContainerAccessPolicyResult describing the container's access policy.
+     * @return A BlobContainerAccessPolicy describing the container's access policy.
      */
-    Azure::Response<Models::GetBlobContainerAccessPolicyResult> GetAccessPolicy(
+    Azure::Response<Models::BlobContainerAccessPolicy> GetAccessPolicy(
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:244]: This one should drop the `Get` prefix, but keep the `Result` suffix.
```suggestion
    Azure::Response<Models::BlobContainerAccessPolicyResult> GetAccessPolicy(
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,14 @@
 - Changed the return type of `StartCopyFromUri` and `StartCopyIncremental` API from a `Response<T>` to the particular `Operation<T>` type called `StartCopyBlobOperation` directly.
 - String conversion functions of extensible enums were renamed from `Get()` to `ToString()`.
 - Moved `SecondaryHostForRetryReads` out of retry options, now it's under `BlobClientOptions`.
+- Changed return type of `BlobServiceClient::GetUserDelegationKey` to `UserDelegationKey`.
+- Changed return type of `BlobServiceClient::GetProperties` to `BlobServiceProperties`.
+- Changed return type of `BlobServiceClient::GetAccountInfo` to `AccountInfo`.
+- Changed return type of `BlobServiceClient::GetStatistics` to `ServiceStatistics`.
+- Changed return type of `BlobContainerClient::GetProperties` to `BlobContainerProperties`.
+- Changed return type of `BlobContainerClient::GetAccessPolicy` to `BlobContainerAccessPolicy`.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:19]:
```suggestion
- Changed return type of `BlobContainerClient::GetAccessPolicy` to `BlobContainerAccessPolicyResult`.
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,14 @@
...
+- Changed return type of `BlobClient::GetProperties` to `BlobProperties`.
+- Changed return type of `BlobClient::GetTags` to `std::map<std::string, std::string>`.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:21]: Are the content-length and content-type headers important for the caller to know about as a first-class scenario?

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp [hunk 4]
```diff
@@ -189,10 +189,10 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
      *
      * @param options Optional parameters to execute this function.
      * @param context Context for cancelling long running operations.
-     * @return A GetDataLakeFileSystemAccessPolicyResult describing the container's access policy.
+     * @return A DataLakeFileSystemAccessPolciy describing the container's access policy.
      * @remark This request is sent to blob endpoint.
      */
-    Azure::Response<Models::GetDataLakeFileSystemAccessPolicyResult> GetAccessPolicy(
+    Azure::Response<Models::DataLakeFileSystemAccessPolciy> GetAccessPolicy(
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:195]: typo:
Policy not Polciy
```suggestion
    Azure::Response<Models::DataLakeFileSystemAccessPolicyResult> GetAccessPolicy(
```

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,10 @@
 - DataLake client constructors won't automatically convert blob url to dfs url anymore.
 - String conversion functions of extensible enums were renamed from `Get()` to `ToString()`.
 - Moved `SecondaryHostForRetryReads` out of retry options, now it's under `DataLakeClientOptions`.
+- Changed return type of `DataLakeServiceClient::GetUserDelegationKey` to `UserDelegationKey`.
+- Changed return type of `DataLakeFileSystemClient::GetProperties` to `DataLakeFileSystemProperties`.
+- Changed return type of `DataLakeFileSystemClient::GetAccessPolicy` to `DataLakeFileSystemAccessPolciy`.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:16]:
```suggestion
- Changed return type of `DataLakeFileSystemClient::GetAccessPolicy` to `DataLakeFileSystemAccessPolicyResult`.
```

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp [hunk 6]
```diff
@@ -52,23 +52,19 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
   using DataLakeSignedIdentifier = Blobs::Models::BlobSignedIdentifier;
   using ListDataLakeFileSystemsIncludeFlags = Blobs::Models::ListBlobContainersIncludeFlags;
 
-  struct GetDataLakeFileSystemAccessPolicyResult
+  struct DataLakeFileSystemAccessPolciy
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp:55]:
```suggestion
  struct DataLakeFileSystemAccessPolicyResult
```

**Comment by ahsonkhan** [comment]
> I didn't change the return type of `BlockBlobClient::GetBlockList` and `PageBlobClient::GetPageRanges`

Makes sense! That was good idea.

However, drop the `Get` prefix from the returned model type names please.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:21]: No, customers don't even have access to the xml body.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:244]: Why? I think AccessPolicy is a resource, like `BlobClient::GetProperties` returns `BlobProperties`.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:19]: duplicat

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp (hunk 4 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:195]: same as in blob, duplicate

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md (hunk 5 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:16]: duplicate

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp (hunk 6 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp:55]: duplicate

**Review by katmsft — APPROVED** [review]
Topic still ongoing, however, the finalized part looks good.

**Comment by Jinming-Hu** [comment]
@ahsonkhan I'll merge this PR first to avoid conflicts with other PRs. I'll follow up on the return type name of Get operations

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp (hunk 1 above)
Comment (ahsonkhan) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:244]: Following this guideline, `AccessPolicy` is different from `GetProperties` and should be a struct wrapper with `Result` suffix. 

> If a request to the service only returns response   headers (and no response body), such as a HEAD request, OR only contains a  response body and “no meaningful response headers”. This becomes the T in the   Response<T>.

From today's discussion return the 4 headers (including etag, lastmodified, etc.):

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp:244]: Then I'm gonna name the return type `GetBlobContainerAccessPolicyResult`, and maybe `BlobContainerAccessPolicyResult` if Ted signs off on your proposal.

**Pull request merged**